### PR TITLE
Task/3948

### LINKED ISSUE #3948: Remove the `attribute/list` call around the codebase
## What is the motivation for adding / enhancing this feature?

If we moved the fetching of attribute meta-data server-side (in the `storefront-api` or `vue-storefront-api` or both) we can potentially resign from fetching attribute metadata client's side). This is a huge optimization area as some of the attributes can have 1000+ different select options - plus some e-shops contains thousands of attributes ...

Then the attribute metadata is being provided to the browser with the `__INITIAL_STATE__` that causes additional performance harm. 

Currently - let's say for a `color` to display it's the value we're using the [`optionLabel`](https://github.com/DivanteLtd/vue-storefront/blob/19a7c2f836209a57089be6dc2d5413c58032c1fc/core/modules/catalog/helpers/optionLabel.ts#L9) helper.

If we add the `custom_attributes` in the `product` entity in the following format:

```json
{ 
  "custom_attributes": [ {
  "attribute_code": "color",
   "value": 150, 
  "label": "Red",
  "is_comparable": true,
  "is_visible": true, 
  "is_user_defined": true
}
]
}
```

... then we don't have to query for this info.

**Note:** To save the product record size we probably should put only the `is_user_defined` and `is_comparable` and `is_visible` attributes on this list

Vue Storefront is expanding the `custom_attributes` and removing the property during the [indexation](https://github.com/DivanteLtd/mage2vuestorefront/blob/51f3b204945e464373829910a837078b4456c156/src/adapters/magento/product.js#L250) and then in the [frontend helper](https://github.com/DivanteLtd/vue-storefront/blob/19a7c2f836209a57089be6dc2d5413c58032c1fc/core/modules/catalog/helpers/index.ts#L475) and [here](https://github.com/DivanteLtd/vue-storefront/blob/19a7c2f836209a57089be6dc2d5413c58032c1fc/core/modules/catalog/helpers/search.ts#L35) and [here](https://github.com/DivanteLtd/vue-storefront/blob/19a7c2f836209a57089be6dc2d5413c58032c1fc/core/modules/catalog/helpers/taxCalc.ts#L141). We should keep this property and potentially stop expanding it in the frontend helpers.

We need to add the `label` property to be set on `mage2vuestorefront` - [in here](https://github.com/DivanteLtd/mage2vuestorefront/blob/51f3b204945e464373829910a837078b4456c156/src/adapters/magento/product.js#L237) as labels don't come from Magento

**Note:** please test how this will work on multi-valued attributes (in that case `value` and `label` should be an array I guess - the `value` is an array 100%)

It's currently expanded for the `configurable_options` - https://github.com/DivanteLtd/vue-storefront/blob/19a7c2f836209a57089be6dc2d5413c58032c1fc/core/modules/catalog/helpers/index.ts#L412

So the only places to remove the attribute label translation are:
- filters,
- product details on PDP.

Then, we can safely remove the `product/list` calls along with the code

## What are the acceptance criteria 

- [ ] ...

## Can you complete this feature request by yourself?

- [ ] YES
- [ ] NO

## Which [Release Cycle](https://docs.vuestorefront.io/guide/basics/release-cycle.html) state this refers to? Info for developer.
Pick one option.

- [x] This is a normal feature request. This should be available on https://test.storefrontcloud.io and then after tests this can be added to next Vue Storefront version. In this case Developer should create branch from `develop` branch and create Pull Request `2. Feature / Improvement` back to `develop`.
- [ ] (**Pick this option only if you're sure**) This is an important improvement request for current Release Candidate version on https://next.storefrontcloud.io and should be placed in next RC version. In this case Developer should create branch from `release` branch and create Pull Request `3. Stabilisation fix` back to `release`.
- [ ] (**Pick this option only if you're sure**) This is a critical improvement request for current Stable version on https://demo.storefrontcloud.io and should be placed in next stable version. In this case Developer should create branch from `hotfix` or `master` branch and create Pull Request `4. Hotfix` back to `hotfix`.

### PR DESCRIPTION
**Pull request by andrzejewsky**
closes #3948
related: https://github.com/DivanteLtd/mage2vuestorefront/pull/99

### CODE REVIEW

File: core/modules/catalog/store/product/actions.ts [hunk 1]
```diff
@@ -455,7 +450,7 @@ const actions: ActionTree<ProductState, RootState> = {
               if (prd.type_id === 'grouped') {
                 subConfigPromises.push(context.dispatch('configureGroupedAsync', prd))
               }
-              subConfigPromises.push(context.dispatch('setupVariants', { product: prd }))
```
Comment (pkarw) [inline core/modules/catalog/store/product/actions.ts]: This is async method shouldn’t we have await here?

File: core/modules/catalog/store/product/actions.ts [hunk 2]
```diff
@@ -493,7 +488,7 @@ const actions: ActionTree<ProductState, RootState> = {
             }
             if (setCurrentProduct || selectDefaultVariant) {
               const subConfigPromises = []
-              subConfigPromises.push(context.dispatch('setupVariants', { product: res }))
+              context.dispatch('setupVariants', { product: res })
```
Comment (pkarw) [inline core/modules/catalog/store/product/actions.ts]: The same question with missing “await”?

File: core/modules/compare/store/actions.ts [hunk 3]
```diff
@@ -7,13 +7,20 @@ import { Logger } from '@vue-storefront/core/lib/logger'
...
-      Logger.info('Compare state loaded from browser cache: ', 'cache', storedItems)()
+      if (storedItems) {
+        commit(types.COMPARE_LOAD_COMPARE, storedItems)
+        Logger.info('Compare state loaded from browser cache: ', 'cache', storedItems)()
+      }
+
+      dispatch(
+        'attribute/loadAttributesFromProducts',
+        { products: getters.getCompareItems },
+        { root: true }
```
Comment (pkarw) [inline core/modules/compare/store/actions.ts]: This is pretty awesome!

**Comment by afirlejczyk** [comment]
One note, vsf-api doesn't return 'options' if it is empty.
response from vsf-api
```JSON
_id: "4928"
_index: "vue_storefront_magento_1579875515"
_score: 0
attributes_metadata: [{
attribute_code: "cost"
attribute_id: 79
default_frontend_label: "Cost"
entity_type_id: 4
frontend_input: "price"
id: 79
is_comparable: false
is_user_defined: true
is_visible: true
is_visible_on_front: false
}]
```

data in ES.
```
{
  "id": 79,
  "default_frontend_label": "Cost",
  "is_visible_on_front": false,
  "is_visible": true,
  "attribute_id": 79,
  "entity_type_id": 4,
  "frontend_input": "price",
  "is_user_defined": true,
  "is_comparable": false,
  "attribute_code": "cost",
  "options": []
},
```

I didn't update vsf-api for some time, so it might be an error only on my side.

**Comment by pkarw** [comment]
@andrzejewsky aren't we merging this in? :)

**Comment by andrzejewsky** [comment]
@pkarw today :)

**Pull request merged**